src/runtime/stream.rs

```diff
@@ -77,21 +84,14 @@ async fn tcp_try_connect(address: &SocketAddr) -> Result<TcpStream> {
     Ok(TcpStream::from_std(std_stream)?)
 }
 
-async fn tcp_connect(address: &ServerAddress) -> Result<TcpStream> {
+pub(crate) async fn tcp_connect(resolved: Vec<SocketAddr>) -> Result<TcpStream> {
```

Comment:
This is just a minor refactor so that the behavior of TCP connection can be tested independently of DNS resolution.